recipes/marisa/all/conanfile.py

```diff
@@ -0,0 +1,150 @@
+import os
+import functools
+from contextlib import contextmanager
+from conans import ConanFile, tools, AutoToolsBuildEnvironment
+from conan.tools.microsoft import is_msvc
+from conan.tools.build import cross_building
+
+required_conan_version = ">=1.45.0"
+
+
+class MarisaConan(ConanFile):
+    name = "marisa"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/s-yata/marisa-trie"
+    description = "Matching Algorithm with Recursively Implemented StorAge "
+    license = ("BSD-2-Clause", "LGPL-2.1")
+    topics = ("algorithm", "dictionary", "marisa")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _use_winbash(self):
+        return tools.os_info.is_windows and (self.settings.compiler == "gcc" or cross_building(self))
```

Comment:
```suggestion
        return self.settings.os == "Windows" and (self.settings.compiler == "gcc" or cross_building(self))
```
os_info is not allowed on CCI. Are people should use dual profile when cross-building, it should be a problem with self.settings.os